File: codeforces/contests/r914d2/C.py
import sys

def main():
    t = int(input().strip())

    for _ in range(t):
        n, k = map(int, input().split())
        arr = list(map(int, input().split()))


        if k >= 3:
            print("0")
        elif k == 1:
            ans = sys.maxsize
            for i in arr:
                ans = min(ans, i)
            for i in range(n):
                for j in range(n):
                    if i == j:
                        continue
                    ans = min(abs(arr[i] - arr[j]), ans)
            print(ans)
        else:
            ans = sys.maxsize
            modf = []
            for i in arr:
                ans = min(ans, i)
            for i in range(n):
                for j in range(n):
                    if i == j:
                        continue
                    ans = min(ans, abs(arr[i] - arr[j]))
                    modf.append(abs(arr[i] - arr[j]))

            for i in range(n):
                for j in range(len(modf)):
                    ans = min(ans, abs(arr[i] - modf[j]))

            print(ans)

if __name__ == "__main__":
    main()

File: codeforces/contests/r914d2/C.cpp
#include <iostream>
#include <cstdint>
#include <climits>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int t;
    cin >> t;

    for (int testcase = 0; testcase < t; ++testcase) {
        int64_t n, k;
        cin >> n >> k;
        vector<int64_t> arr(n);

        for (int64_t i = 0; i < n; ++i) {
            cin >> arr[i];
        }

        int64_t ans = LLONG_MAX;
	for (int64_t i : arr) {
                ans = min(ans, i);
        }
        if (k >= 3) {
            cout << "0" << endl;
        } else if (k == 1) {
            for (int64_t i : arr) {
                ans = min(ans, i);
            }
            for (int64_t i = 0; i < n; ++i) {
                for (int64_t j = 0; j < n; ++j) {
                    if (i == j) {
                        continue;
                    }
                    ans = min(abs(arr[i] - arr[j]), ans);
                }
            }
            cout << ans << endl;
        } else {
            vector<int64_t> modf;
            for (int64_t i : arr) {
                ans = min(ans, i);
            }
	    sort(arr.begin(), arr.end());
            for (int64_t i = 0; i < n; ++i) {
                for (int64_t j = 0; j < n; ++j) {
                    if (i == j) {
                        continue;
                    }
                    ans = min(ans, abs(arr[i] - arr[j]));
		    auto l = lower_bound(arr.begin(), arr.end(), abs(arr[i]-arr[j])) - arr.begin();
		    if (l == 0) ans = min(ans, abs(arr[1] - abs(arr[i] - arr[j])));
		    else if (l == arr.size()) ans = min(ans, abs(arr[n-1] - abs(arr[i] - arr[j])));
		    else
		    {
			ans = min(ans, abs(arr[l-1] - abs(arr[i] - arr[j])));
			ans = min(ans, abs(arr[l+1] - abs(arr[i] - arr[j])));
			ans = min(ans, abs(arr[l] - abs(arr[i] - arr[j])));
		    }
                }
            }


            cout << ans << endl;
        }
    }

    return 0;
}

